### PR TITLE
Introduce `enum`, `newtype`, and `:check-docstrings`

### LINKED ISSUE #244: Introduce `enum` and perhaps `newtype` toward the end of the Language Basics lab.

### PR DESCRIPTION
**Pull request by WeeknightMVP**
Fixes #244.

### CODE REVIEW

**Comment by WeeknightMVP** [comment]
TODO: Break `Language Basics` into multiple smaller labs.

File: labs/Language/Basics.md [hunk 1]
```diff
@@ -149,16 +149,28 @@ base 10.
...
-must be `module Path::...Path::ModuleName where`. The `Path::...Path` component
-is the system path from the root of whatever set of modules you're
-creating or working from. The `ModuleName` component is the basename
-of this file. For instance, this module is `labs::Language::Basics`
-because its path from the root repository is
-`labs/Language/Basics.md`. There's really not much to naming
-modules. But don't forget the `where` clause at the end.
-
-Importing modules is also pretty simple. Just add a line starting with
+This file is a Cryptol module. The first interpreted* line of every
```
Comment (weaversa) [inline labs/Language/Basics.md:152]: What is the `*` on `interpreted*` for?

File: labs/Language/Basics.md [hunk 2]
```diff
@@ -26,7 +26,7 @@ Specifically, you'll also gain experience with
   * type parameters and type constraints,
   * the `:check`, `:prove`, and `:sat` commands,
```
Comment (weaversa) [inline labs/Language/Basics.md:27]: add `:check-docstrings`

**Review by weaversa — APPROVED** [review]

File: labs/Language/Basics.md [hunk 3]
```diff
@@ -363,18 +392,43 @@ Other data types include:
...
 
-  * Integers modulo _n_: Each type of the form `[n]`, described above, provides a 
-    [least residue
-    system](https://en.wikipedia.org/wiki/Modular_arithmetic#Residue_systems)
-    for [integers modulo
-    2<sup>n</sup>](https://en.wikipedia.org/wiki/Modular_arithmetic#Integers_modulo_n).
-    Types of the form `Z n` provide a least residue system for any positive _n_. E.g.,
-    `4 + 4 : Z 7` evaluates to `1`.
+  * `newtype`s, which are named types defined by records, e.g.
+    ``; this new type is treated distinctly from any other types and
```
[the remark this answers is not in the document]
Comment (WeeknightMVP) [reply labs/Language/Basics.md:396]: ...yes.

File: labs/Language/Basics.md (hunk 2 above)
Comment (WeeknightMVP) [reply labs/Language/Basics.md]: Done!

File: labs/Language/Basics.md (hunk 1 above)
Comment (WeeknightMVP) [reply labs/Language/Basics.md]: Took out the confusing asterisk.

**Pull request merged**